src/common/grpc_client.rs

```diff
@@ -162,21 +162,20 @@ pub struct GrpcClient<'a> {
 
 impl<'a> Drop for GrpcClient<'a> {
     fn drop(&mut self) {
-        if let Some(task) = self.http2_task.take() {
-            //(TODO(RSDK-3061)) avoid blocking on if possible
-            block_on(self.executor.run(async {
-                task.cancel().await;
-            }));
-        }
+        let _ = self.http2_task.take();
```

Comment:
Why is it no longer necessary to wait on task cancelation here?

Reply:
We don't need it really (because the TCP layer will close the connection gracefully) But better to add so we close the HTTP2 connection a bit more gracefully